### PR TITLE
Fix #119

### LINKED ISSUE #119: Infinite loop in OpenAL GetStringList
The loop in `GetStringList` will never break because `Marshal.PtrToStringAnsi` only returns null when the pointer passed into it is null.

A better alternative might be to check if the string is null or if its length is 0.

### PR DESCRIPTION
**Pull request by Perksey**
# Summary of the PR
Fixes #119 

# Related issues, Discord discussions, or proposals
Links go here.

# Further Comments

### CODE REVIEW

**Comment by Perksey** [comment]
This is ready for review.

@waldnercharles is this sufficient for your needs?

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Creative/EnumerateAll.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public IEnumerable<string> GetStringList(GetEnumerateAllContextStringList param)
                 var currentPos = result;
                 while (true) {
                     var currentString = Marshal.PtrToStringAnsi(new IntPtr(currentPos));
-                    if (currentString is null) {
+                    if (!string.IsNullOrEmpty(currentString)) {
```
Comment (Pyrdacor) [inline src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Creative/EnumerateAll.cs]: The logic is inversed now. I think you have to remove the negation.

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.EXT/CaptureEnumerationEnumeration.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public IEnumerable<string> GetStringList(GetCaptureContextStringList param)
                 var currentPos = result;
                 while (true) {
                     var currentString = Marshal.PtrToStringAnsi(new IntPtr(currentPos));
-                    if (currentString is null) {
+                    if (!string.IsNullOrEmpty(currentString)) {
```
Comment (Pyrdacor) [inline src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.EXT/CaptureEnumerationEnumeration.cs]: Again wrong negation here.

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Enumeration/Enumeration.cs [hunk 3]
```diff
@@ -43,7 +43,7 @@ public IEnumerable<string> GetStringList(GetEnumerationContextStringList param)
                 var currentPos = result;
                 while (true) {
                     var currentString = Marshal.PtrToStringAnsi(new IntPtr(currentPos));
-                    if (currentString is null) {
+                    if (!string.IsNullOrEmpty(currentString)) {
```
Comment (Pyrdacor) [inline src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Enumeration/Enumeration.cs]: Also wrong negation

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Creative/EnumerateAll.cs (hunk 1 above)
Comment (Perksey) [reply src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Creative/EnumerateAll.cs]: Ah thanks!

Comment (Perksey) [reply src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Creative/EnumerateAll.cs]: Fixed.

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Enumeration/Enumeration.cs (hunk 3 above)
Comment (Perksey) [reply src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.Enumeration/Enumeration.cs]: Fixed.

File: src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.EXT/CaptureEnumerationEnumeration.cs (hunk 2 above)
Comment (Perksey) [reply src/OpenAL/Extensions/Silk.NET.OpenAL.Extensions.EXT/CaptureEnumerationEnumeration.cs]: Fixed.

**Pull request merged**